sdk/core-rest/core-client/review/core-client.api.md

```diff
@@ -79,7 +79,7 @@ export type RequestParameters = {
 };
 
 // @public
-export type RouteParams<TRoute extends string> = TRoute extends `{${infer _Param}}/${infer Tail}` ? [pathParam: string, ...pathParams: RouteParams<Tail>] : TRoute extends `{${infer _Param}}` ? [pathParam: string] : TRoute extends `${infer _Prefix}:${infer Tail}` ? RouteParams<`{${Tail}}`> : [];
+export type RouteParams<TRoute extends string> = TRoute extends `${infer _Head}/{${infer _Param}}${infer Tail}` ? [pathParam: string, ...pathParams: RouteParams<Tail>] : [];
```

Comment:
@willmtemple do you have enough TS wisdom to review this part of the magic?

Reply:
Added some comments on what this type is doing